### ISSUE TITLE
What happens with result publication and the COVID WAM rule?

### ISSUE THREAD

**Issue by matomatical** [matomatical/UoM-WAM-Spam#24]
How do marks excluded from the WAM calculation due to COVID appear on the results page, and how does this app see them? Will it complicate any future attempts at solving #17?

I don't have any results from this semester yet, and none from last semester, so if someone already knows what this all looks like on the results page, please feel free to fill me in. Alternatively, I'm sure I'll be able to see for myself what a lower-than-current-WAM result looks like some time in the next few days.

**Comment by zcduthie** [comment]
Not totally across the details - but can confirm this is what currently sits in my results.txt (made up results):
(Started degree at beginning of 2020. Completed two subjects in Semester 1 and no WAM yet)

```
"Master of Information Technology": {
    "wam": null,
    "results": [
      {
        "subject": "COMP90041 Programming and Software Development",
        "date": "2020, Semester 1",
        "mark": "95",
        "grade": "H1^",
        "credits": "12.500"
      },
      {
        "subject": "INFO90002 Database Systems & Information Modelling",
        "date": "2020, Semester 1",
        "mark": "95",
        "grade": "H1^",
        "credits": "12.500"
      }
    ]
  },
```

Let me know if there's something particular that'll be useful to check.

**Comment by matomatical** [comment]
Thanks for this sample.

It's interesting that you have no WAM yet. When I completed the first two subjects of my Master of Science (CS) in 2019, I certainly got a WAM. Maybe there's part of the COVID rules for new students' WAM calculation which I missed (probably because it wouldn't apply to me), which is why your average has not been calculated yet?

As for the subjects, that '^' character next to the grade must come from the COVID rule. I guess it means 'this result was excluded from the WAM calculation'. My results.txt has grades without '^' (from 2019 and earlier) and this caret symbol is the symbol that I understand to be used on official transcripts for the same purpose.

In conclusion, I suppose 2020 does indeed complicate #17. That project will need to take into account that 2020 subjects are only included in certain situations when it is reverse-engineering the WAM calculation. At least all the necessary information is contained in the data we already record :).

**Comment by matomatical** [comment]
To complete the picture it seems:

* This semester, the results are released regardless of the current WAM, and the WAM is updated,
* The '^' and revision to the WAM will apparently happen at a later date.

Assuming the COVID rule disappears after this semester, this might not be relevant for #17 if we add that feature in a future semester (it's getting late into the results period now).
